server/data/auth/authenticate.js

```diff
@@ -0,0 +1,43 @@
+
+//Adding jwt token 
+const jwt = require('jsonwebtoken')
+
+const jwtKey =
+process.env.JWT_SECRET || "supercalifragilisticexpialidocious"
+
+
+// quickly see what this file exports
+module.exports = {
+    authenticate,
+    generateToken
+  };
+
+// To authenticate the user  
+function authenticate (req, res, next)  {
+    const token = req.get("Authorization")
+
+    if(token) {
+        jwt.verify(token, jwtKey, (err, decoded) => {
+            if(err) return res.status(401).json(err)
+            req.decoded = decoded 
+            next()
+        })
+    } else {
+        return res.status(401).json({
+            error: "No token provided, must be set on Authorization Header"
+        })
+    }
+}
+
+// this will generate the token for the user 
+function generateToken(user){
+    //get the username 
+    const payload ={
+        username: user.username
+    }
+    //Expire the user in an hour
+    const options = {
+        expiresIN: '1hr'
```

Comment:
Should be `expiresIn` as per [NPM jsonwebtoken](https://www.npmjs.com/package/jsonwebtoken). `jwtid` isn't required but I think we should have it.

Reply:
jwtKey is required as it sets the key for the token. You need the key in order to create the token if you don't have the key the token will not generate. At least the last time I did this I needed the key, otherwise it won't work ```
  const jwtKey =
process.env.JWT_SECRET || "supercalifragilisticexpialidocious"
```

Expires In means their token will expire in that amount of time. We can always change this but it is good to have a time limit on their login so that someone is not logged in all the time. Less security risk. And they will definitely get a new token. I would like to go further and somehow get it to generate a new toke for different platforms.

Reply:
👍 

With `expiresIn`, I was referring to the capitalization of `IN`. I believe the `n` should be lowercase.